src/main/java/io/nats/client/api/BucketConfiguration.java

```diff
@@ -0,0 +1,292 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package io.nats.client.api;
+
+import java.time.Duration;
+
+import static io.nats.client.support.NatsKeyValueUtil.*;
+import static io.nats.client.support.Validator.*;
+
+/**
+ * The BucketConfiguration class contains the configuration for of a Key Value bucket.
+ */
+public class BucketConfiguration {
+    private final StreamConfiguration sc;
+    private final String name;
+
+    static BucketConfiguration instance(String json) {
+        return new BucketConfiguration(StreamConfiguration.instance(json));
+    }
+
+    BucketConfiguration(StreamConfiguration sc) {
+        this.sc = sc;
+        name = extractBucketName(sc.getName());
+    }
+
+    public StreamConfiguration getBackingConfig() {
+        return sc;
+    }
+
+    /**
+     * Gets the name of this bucket.
+     * @return the name of the bucket.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the maximum number of values across all keys, including history values
+     * @return the maximum number of values for the bucket
+     */
+    public long getMaxValues() {
+        return sc.getMaxMsgs();
+    }
+
+    /**
+     * Gets the maximum number of history for any one key. Includes the current value.
+     * @return the maximum number of values for any one key.
+     */
+    public long getMaxHistory() {
+        return sc.getMaxMsgsPerSubject();
+    }
+
+    /**
+     * Gets the maximum number of bytes for this bucket.
+     * @return the maximum number of bytes for this bucket.
+     */
+    public long getMaxBucketSize() {
+        return sc.getMaxBytes();
+    }
+
+    /**
+     * Gets the maximum number of bytes for an individual value in the bucket.
+     * @return the maximum bytes for a value.
+     */      
+    public long getMaxValueSize() {
+        return sc.getMaxMsgSize();
+    }
+
+    /**
+     * Gets the maximum age for a value in this bucket.
+     * @return the maximum age.
+     */
+    public Duration getTtl() {
+        return sc.getMaxAge();
+    }
+
+    /**
+     * Gets the storage type for this bucket.
+     * @return the storage type for this stream.
+     */
+    public StorageType getStorageType() {
+        return sc.getStorageType();
+    }
+
+    /**
+     * Gets the number of replicas for this bucket.
+     * @return the number of replicas
+     */    
+    public int getReplicas() {
+        return sc.getReplicas();
+    }
+
+    /**
+     * Gets the duplicate checking window bucket. Duration.ZERO
+     * means duplicate checking is not enabled.
+     * @return the duration of the window.
+     */    
+    public Duration getDuplicateWindow() {
+        return sc.getDuplicateWindow();
+    }
+
+    @Override
+    public String toString() {
+        return "BucketConfiguration{" +
+                "name='" + name + '\'' +
```

Comment:
The name is validated using `validatePrintableExceptWildDotGt` which is not what a valid KV bucket name is, it should be `\A[a-zA-Z0-9_-]+\z`.

Inherently if `validatePrintableExceptWildDotGt` does what it says then name can have a `"` in it which would produce invalid JSON

Reply:
okay yeah, I didn't get to watches, I've planned this for later. 
I will fix validations, good catch